iothub_client/samples/iothub_client_sample_upload_to_blob_custom/iothub_client_sample_upload_to_blob_custom.c

```diff
@@ -0,0 +1,154 @@
+// Checking of return codes and error values shall be omitted for brevity.  Please practice sound engineering practices
+// when writing production code.
+
+#ifndef DONT_USE_UPLOADTOBLOB
+
+#include <stdio.h>
+#include <stdlib.h>
+
+/* This sample uses the _LL APIs of iothub_client for example purposes.
+That does not mean that HTTP only works with the _LL APIs.
+Simply changing the using the convenience layer (functions not having _LL)
+and removing calls to _DoWork will yield the same results. */
+
+#include "iothub.h"
+#include "iothub_device_client.h"
+
+#include "azure_c_shared_utility/shared_util_options.h"
+#include "iothub_message.h"
+#include "iothubtransportmqtt.h"
+
+#ifdef SET_TRUSTED_CERT_IN_SAMPLES
+#include "certs.h"
+#endif // SET_TRUSTED_CERT_IN_SAMPLES
+
+/*String containing Hostname, Device Id & Device Key in the format:                         */
+/*  "HostName=<host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>"                */
+/*  "HostName=<host_name>;DeviceId=<device_id>;SharedAccessSignature=<device_sas_token>"    */
+static const char* connectionString = "[device connection string]";
+
+/*Optional string with http proxy host and integer for http proxy port (Linux only)         */
+static const char* proxyHost = NULL;
+static int proxyPort = 0;
+
+static const char* data_to_upload_format = "Hello World from iothub_client_sample_upload_to_blob_custom: %d\n";
+static char data_to_upload[128];
+
+int main(void)
+{
+    IOTHUB_DEVICE_CLIENT_LL_HANDLE device_ll_handle;
+
+    // Used to initialize IoTHub SDK subsystem
+    (void)IoTHub_Init();
+    (void)printf("Starting the IoTHub client sample upload to blob...\r\n");
+
+    device_ll_handle = IoTHubDeviceClient_LL_CreateFromConnectionString(connectionString, MQTT_Protocol);
+    if (device_ll_handle == NULL)
+    {
+        (void)printf("Failure creating IotHub device. Hint: Check your connection string.\r\n");
+    }
+    else
+    {
+#ifndef WIN32
+        size_t log = 1;
+        (void)IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_CURL_VERBOSE, &log);
+#endif // !WIN32
+
+#ifdef SET_TRUSTED_CERT_IN_SAMPLES
+        // Setting the Trusted Certificate. This is only necessary on systems without
+        // built in certificate stores.
+        IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_TRUSTED_CERT, certificates);
+#endif // SET_TRUSTED_CERT_IN_SAMPLES
+
+        HTTP_PROXY_OPTIONS http_proxy_options = { 0 };
+        http_proxy_options.host_address = proxyHost;
+        http_proxy_options.port = proxyPort;
+
+        if (proxyHost != NULL && IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_HTTP_PROXY, &http_proxy_options) != IOTHUB_CLIENT_OK)
+        {
+            (void)printf("failure to set proxy\n");
+        }
+        else
+        {
+            char* uploadCorrelationId;
+            char* azureBlobSasUri;
+            const char* bla = "subdir/hello_world_custom_mb.txt";
```

Comment:
nit: more descriptive name here would be helpful

Reply:
Gahhh! I forgot that one.

Reply:
Fixed.